### DISCUSSION TITLE
Differences Between Journal Article and Current Software

### DISCUSSION THREAD

**Discussion by DarioS** [vdemichev/DiaNN#142] [General]
I notice that the Top 3 Sum method is used for generating protein-level quantities in the journal article but the website mentions maxLFQ a lot but it's entirely missing in the journal article. Is there a reason why it is different? Can I confirm that the Normalised column is the traditional Top 3 Sum and doesn't change as more samples are added to a project, whereas MaxLFQ column would change as the project grows with more samples? I'm confused why the column keywords in Main Report section do not appear in Main Output Reference. Aren't they supposed to be a subset of the full reference? Also, I wonder about getting peptide-level data from DIA-NN. How should it be done? I know there is the hassle of largely overlapping peptide sequences because of missed trypsin cleavages to deal with and other considerations. For example,

```
YLAEVACGDDRK
|||||||||||•
YLAEVACGDDR
```

**Comment by vdemichev** [comment]
Hi Dario,

The Nature Methods paper is old, DIA-NN has been improved tremendously since then. We now recommend using MaxLFQ.
Normalised column is Top 1, which performs better than Top 3.
All Normalised columns will change as samples are added, unless normalisation is disabled.
Note that you can apply MaxLFQ manually using diann R package or the iq R package. Can also do it for each batch separately,  allowing to add batches to the project without changing quantities calculated previously. Please see details on incremental data processing here https://github.com/vdemichev/DiaNN#frequently-asked-questions.

There are some columns in the main report which are not described because they are not supposed to be used by the end-user.

Please see an example on how to get peptide quantities here: https://github.com/vdemichev/diann-rpackage

Best,
Vadim

**Reply by DarioS** [reply]
Thanks. Could you elaborate about what cross-run normalisation does? My understanding is that the total sum of precursor ions is calculated for each sample and a multiplication factor is used to multiply each sample so that the sum of every sample is the same. If I set sample A to be the reference sample (i.e. gets multiplied by 1) and I rescale samples B and C by some multiplier (e.g. A by 1.22, B by 0.93), I don't understand how adding sample D to the data set would require B and C to be rescaled. I am missing something. On another note, do you have updated yeast spike-in benchmarking results for Figure 2 using MaxLFQ?

**Reply by vdemichev** [reply]
Please see details on the normalisation in the DIA-NN paper. The recent versions works similarly, just does so in RT-dependent fashion, to compensate for potential issues with peptide purification that might affect hydrophilic and hydrophobic peptides differently.

**Reply by vdemichev** [reply]
Yes, if MaxLFQ is used for LFQbench-type of benchmarks, it significantly improves quantitative performance. One example is our dia-PASEF preprint https://www.biorxiv.org/content/10.1101/2021.03.08.434385v1

**Reply by DarioS** [reply]
Ah, I have read about cross-run normalisation and now understand that each sample would depend on all other samples because q-values are calculated in a cross-run manner ("that is, the minimum across all the acquisitions"). Therefore, as more samples are added, different precursor ions become included which were excluded before, if the minimum q-value is lower than the threshold. So, the TopN precursor ions might change as the precusors with signficant q-values change. I hope that is correct. Also, the top 3 average correlated fragments might change as more samples are added and the correlation of each fragment changes, potentially changing the rankings.

I wonder about Main Report data and its correspondence to the steps described.

- It has a column Genes.MaxLFQ but the DIA-NN companion R package and iq are recommended for LFQ in the user giude. I am confused why the two R varieties are recommended as a post-hoc processing step rather than the in-built C/C++ version, other than for large batches over time. Are they functionally equivalent otherwise to the in-built variety?
- There are six fragments in the Quant.Raw and Quant.corrected columns. and sometimes they have identical values. For example, for AAADALSDLEIK2. the fifth fragment of sample _a_ is 185.665 in both raw and uncorrected columns, but the other five fragments change value between raw and corrected. Should all six of the Top 6 Fragments be corrected? Fragment 5 is the highest correlated of the Top 3 correlated fragments used for precursor ion estimation, so it seems strange that correction is only applied to only 2 out of 3 of the Top 3 fragments and one uncorrected fragment is used to estimate the precursor intensity.
- I also wonder about precursor normalisation: "Precursors are then ordered by their coefficients of variation. Top pN precursors are selected". You implicitly mean ordered in increasing order of CV, so that the ones with smallest CV are chosen for normalisation, right? Each total sum calculated is one number per bioloigical sample of all of the precursors belonging to any protein and not one sum per peptide, isn't it? So you multiply each sample by one number, correct? Also, what is the default value of p in the software and how can the user change it?
- Genes.Normalised column is equivalent to Top 1 precursor, isn't it?
- "When imputation is necessary, we prefer to perform it on the protein level." Fragment.Quant.Raw column of the example report CSV file I downloaded appears to have no missing values (always 6 numbers). What happened to missing values?

Have you considered illustrating the workflow in a flowchart format which shows how each stage corresponds to the different kinds of colmns output? It would be greatly beneficial to understanding how the method works from a novice perspective.

**Reply by vdemichev** [reply]
- "but the DIA-NN companion R package and iq are recommended for LFQ in the user giude" - only if you want to do batch-correction/filtering on the peptide level, or if protein quantities calculated by DIA-NN itself are not suitable for some other reason. No, they are not functionally equivalent, the built-in algorithm is a bit (just a bit) more advanced.

- If corrected quantity is the same as uncorrected, means no interferences were detected and subtracted for the this fragment.

- Yes, ordered by CVs. Yes, correct about the total sum. Yes, one number (it's global normalisation): more advanced RT-dependent normalisation is not described in the paper. p is 0.4, can change it using --norm-fraction [p]. 

- Yes

- Main report (precursor level) simply does not list missing IDs.

- Yes, there's a simple flowchart in the DIA-NN paper. I agree would be nice to have one for the current (a lot more advanced workflow). Thank you for the suggestion!

**Reply by DarioS** [reply]
Thanks for all of the clarification. A pretty flowchart will be much appreciated. Two more questions for you.

- Is there any normalisation already applied to Fragment.Quant.Raw or would it be a good starting point for applying methods?
- Is one type of fragment noise corrected caused by different peptides with permuted amino acids e.g. PEPTIDE vs. PEPDITE?

**Reply by vdemichev** [reply]
No, Fragment.Quant.Raw is contains integrated chromatograms without any normalisation. 
I would think that in most cases aa permutation would cause a retention time shift, hence these peptides will not interfere. Rather interference comes from completely unrelated peptides (which might share some fragment masses though).
